dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DataControllerTest.kt

```diff
@@ -115,19 +118,60 @@ class DataControllerTest {
                 assertDoesNotThrow {
                     apiAccessor.companyRolesControllerApi.hasUserCompanyRole(role, companyId, dataReaderUserId)
                 }
-                assertDoesNotThrow { uploadEuTaxoDataset(companyId) }
+                assertDoesNotThrow { uploadEuTaxoDataset(companyId, false) }
             } else {
-                assertAccessDeniedWrapper { uploadEuTaxoDataset(companyId) }
+                assertAccessDeniedWrapper { uploadEuTaxoDataset(companyId, false) }
             }
         }
     }
 
-    private fun uploadEuTaxoDataset(companyId: UUID) {
+    @Test
+    fun `assure that bypassQa is allowed for Dataland Keycloak admin or reviewer`() {
+        val companyId = UUID.fromString(
+            apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId,
+        )
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.assignCompanyRole(CompanyRole.DataUploader, companyId, dataReviewerUserId)
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reviewer)
+        assertDoesNotThrow { uploadEuTaxoDataset(companyId, true) }
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        assertDoesNotThrow { uploadEuTaxoDataset(companyId, true) }
+    }
+
+    @Test
+    fun `assure that bypassQa is forbidden for users unless they are a company owner`() {
```

Comment:
Testbeschreibung passt noch nicht ganz, da das ja auch für keycloak admin und reviewer möglich ist. Vll etwas in die Richtung: assure bypassQA is forbidden for users with reader rights unless they have company role companyowner

Reply:
ok